frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/common/useEditFieldForm.ts

```diff
@@ -0,0 +1,93 @@
+import { useMemo } from 'react'
+import {
+  DeepPartial,
+  UnpackNestedValue,
+  useForm,
+  UseFormReturn,
+} from 'react-hook-form'
+import { useDebounce } from 'react-use'
+
+import { FieldBase } from '~shared/types/field'
+
+import { EditFieldProps } from './types'
+import { getButtonText } from './utils'
+
+type UseEditFieldFormProps<
+  FormShape,
+  FieldShape extends FieldBase,
+> = EditFieldProps<FieldShape> & {
+  transform: {
+    input: (field: FieldShape) => FormShape
+    output: (form: FormShape, originalField: FieldShape) => FieldShape
+  }
+}
+
+type UseEditFieldFormReturn<U> = UseFormReturn<U> & {
+  handleUpdateField: () => Promise<void>
+  buttonText: string
+  isSaveEnabled: boolean
+}
+
+export const useEditFieldForm = <FormShape, FieldShape extends FieldBase>({
+  field,
+  isPendingField,
+  handleChange,
+  handleSave,
+  transform,
+}: UseEditFieldFormProps<
+  FormShape,
+  FieldShape
+>): UseEditFieldFormReturn<FormShape> => {
+  const defaultValues = useMemo(
+    () => transform.input(field) as UnpackNestedValue<DeepPartial<FormShape>>,
+    [field, transform],
+  )
+  const editForm = useForm<FormShape>({
+    defaultValues,
+  })
+
+  const watchedInputs = editForm.watch()
+
+  useDebounce(
+    () => handleChange(transform.output(watchedInputs as FormShape, field)),
+    300,
+    Object.values(watchedInputs),
+  )
+
+  const handleUpdateField = editForm.handleSubmit((inputs) => {
+    const updatedFormField: FieldShape = transform.output(
+      inputs as FormShape,
+      field,
+    )
+    return handleSave(updatedFormField, {
+      onSuccess: (newField) => {
+        editForm.reset(
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          transform.input(newField),
+          // react-hook-form has strange behaviour where setting a field
+          // to undefined results in the form displaying the field in
+          // a previous state. Hence don't reset the form values.
```

Comment:
`reset` resets the form to its initial state with `defaultValues`, which has been populated above; which may be why the field is seemingly set to the previous state. May just need to reset everything to empty strings instead of `undefined`?.

Reply:
thanks, empty string seems to work! [1ea6867](https://github.com/opengovsg/FormSG/pull/3581/commits/1ea686727ac12b6d0ec500bed7c812a7852f74e1)